tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md

```diff
@@ -0,0 +1,606 @@
+
+    ```Javascript
+    opaTest('#1: ListReport: Check List Report Page loads', function(Given, When, Then) {
+        When.onTheMainPage.onFilterBar().iExecuteSearch();
+        Then.onTheMainPage.onTable().iCheckRows(5);
+    });
+    ```
+
+    To perform the required check whether specific values are available in the fields of a list report table, you can use the test function `iCheckRows()`. To do so, pass a condition as a parameter to identify a value within a column of the table. In the example below, column **Travel** is checked for value **2**.
+
+    ```Javascript
+    ...
+      Then.onTheMainPage.onTable()
+          .iCheckRows(5)
+          .and.iCheckRows({ "Travel": "2" });
+    });
+    ```
+
+    You can check connected description/ID field values by passing the complete content to the test function. In the example below, the content **Ryan (594)** is checked in the column **Customer**.
+
+    ```Javascript
+    ...
+      Then.onTheMainPage.onTable()
+          .iCheckRows(5)
+          .and.iCheckRows({ "Travel": "2" })
+          .and.iCheckRows({ "Customer": "Ryan (594)" });
+    });
+    ```
+
+2. Load filtered data into the list report table
+
+    Load filtered data into the list report to avoid too much data being loaded. You can use the filter bar to do this. You can set values in the filter fields with the test function `iChangeFilterField()`. Add the coding below right at the end of test 1 to set a filter field and to reload the data.
+
+    ```Javascript
+    ...
+      When.onTheMainPage.onFilterBar()
+          .iChangeFilterField("Status", "A")
+          .and.iExecuteSearch();
+      Then.onTheMainPage.onTable()
+          .iCheckRows(1);
+    });
+    ```
+
+    The filter field **Status** is set to value **A** for "Accepted" to load all records with that status. Additionally, another assertion is entered to check that exactly one record is loaded with this filter.
+
+
+    To again load all available data, clear the filter field by calling the same function with an empty string as the value. Reloading the data shows 5 records.
+
+    ```Javascript
+    ...
+      When.onTheMainPage.onFilterBar()
+          .iChangeFilterField("Status", "", true)
+          .and.iExecuteSearch();
+      Then.onTheMainPage.onTable().iCheckRows(5);
+    });
+    ```
+
+3. Execute an action on a selected item in the list report
+
+    Before executing standard or custom bound actions, you can select an item in the list report table by using the test function `iSelectRows()`. As with the check function for rows, you can determine the selection by a condition. The action to be executed will be the deletion of a travel item. Do this by calling the test function `iExecuteDelete()`.
```

Comment:
"custom bound actions" - This one may also need hyphens.

Reply:
done